Shared/Samples/Filter building scene layer/FilterBuildingSceneLayerView.BuildingGroupSublayerToggleView.swift

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension FilterBuildingSceneLayerView {
+    /// The building group sublayer toggle which has a toggle for the visbility of the group
+    /// and each of its sublayers.
+    struct BuildingGroupSublayerToggleView: View {
+        /// A Boolean value indicating if the group sublayer is visible.
+        @State private var isVisible: Bool
+        
+        /// The group sublayer which is used to build this view.
+        let groupSublayer: BuildingGroupSublayer
+        
+        /// Creates the custom toggle view with a group sublayer.
+        /// - Parameter groupSublayer: The group sublayer which is used to build this view.
+        init(groupSublayer: BuildingGroupSublayer) {
+            // Sets the initial value of the toggle to the
+            // current visbility of the group sublayer.
+            isVisible = groupSublayer.isVisible
+            
+            self.groupSublayer = groupSublayer
+        }
```

Comment:
It is best not to set a state property in an initializer unless absolutely necessary. Please remove this initializer and instead update `isVisible` in an `onAppear` modifier.

Reply:
They shouldn't see the toggle change either way since [`onAppear`](https://developer.apple.com/documentation/swiftui/view/onappear(perform:))'s action is called before the view actually appears on screen: "The exact moment that SwiftUI calls this method depends on the specific view type that you apply it to, but the action closure completes before the first rendered frame appears."

Reply:
Good thought. It could have been that.